### PR TITLE
[FRONT] Add Beehive Create And Update Forms

### PR DESCRIPTION
**Pull request by amakhajar**
## Info
This PR is based on https://hivefive.atlassian.net/browse/HV-55

## Features

- BeehiveCreate Page: Form to add a new beehive to the existing apiary
- BeehiveUpdate Page: Form to update existing beehive
- BeehiveForm : Component to create a generic Form for create and update actions
- Added the needed translations 

### Adding a new beehive test
1.  Go to http://127.0.0.1:63281/apiary/list
2.  Click on a random Apiary box. This will redirect you to a list of Beehives in this Apiary
image

4.  This Apiary for example has no Beehives for now. Click on the "+" icon. You will be redirected to the following page.
image

> You can make sure from the URL that you are adding a beehive to the chosen apiary

5.   Fill in the form with the needed information 
image
6.   Click on "Ajouter la ruche" button and **wait** for the confirmation before leaving this page
image
7.   Make sure that the Beehive was added by clicking on the "Go back" arrow on top left
image
8.   Click on the Beehive box and you will be redirected to the Beehive home page
image

### Edit an existing beehive test

1.  Once on the Beehive Home page, click on the three dots on top right corner (Delete Page is not yet working)
image
2.  Click on "Modifier" and update the informations on the form 
image
3.  Click on "Modifier la ruche" button and **wait** for the confirmation before leaving this page 
image
4.  Go back to Beehive Home and make sure the data was updated 
image

> Note: You need to refresh Beehive Home and Apiary Home after each action! otherwise you won't be able to see your updates

### CODE REVIEW

File: front/src/components/BeeHive/BeehiveForm.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+export default function BeehiveForm({
+  onSubmit,
+  initialData = undefined,
+  errors = {},
+}: CreateProps | UpdateProps) {
+
+  const { uid } = useParams();
+  const isUpdate = initialData !== undefined;
+  const { previousUrl } = usePreviousUrlFromLocation();
+  const beeTypes = ['Black', 'Italian', 'Caucasian', 'Carnolien', 'Buckfast'];
```
Comment (Spaicat) [inline front/src/components/BeeHive/BeehiveForm.tsx]: Should pull this from the backend, not hardcode it.
I believe the function is called `listBeeType` in the backend, you should look into this.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (Spaicat) [inline .vscode/settings.json]: Remove this useless file

File: front/src/components/BeeHive/BeehiveForm.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  return <form className="mb-10 max-w-screen-lg" onSubmit={submit}>
+    {
+      !isUpdate ?
+        <TopNavigationMenu
+          previousPath={previousUrl ?? (uid ? route(ApiaryHome, { uid }) : route(ApiaryList))} /> :
+        <TopNavigationMenu
+          previousPath={previousUrl ?? (uid ? route(BeehiveHome, { uid }) : route(ApiaryList))} />
+    }
```
Comment (Spaicat) [inline front/src/components/BeeHive/BeehiveForm.tsx]: Is this better? (For not repeating too much stuff)
```suggestion
    <TopNavigationMenu
      previousPath={previousUrl ??
        (uid
          ? route(!isUpdate ? ApiaryHome : BeehiveHome, { uid })
          : route(ApiaryList))}
    />
```

**Review by Spaicat — APPROVED** [review]

File: front/package-lock.json
Comment (rphang) [inline front/package-lock.json:1]: This file shouldn't be removed. It may be updated if new packages are added or updated.

**Review by rphang — DISMISSED** [review]

File: front/package-lock.json
Comment (amakhajar) [reply front/package-lock.json:1]: windaube :')Promise I didn't remove it on my own. I'll fix it tomorrow

Comment (rphang) [reply front/package-lock.json:1]: Nvm! It seems that the newly downloaded packages have some regression on their dependencies, my bad! LGTM.

**Pull request merged**